### PR TITLE
[BEAM-13884] Improve mtime package

### PR DESCRIPTION
**Pull request by jrmccluskey**
Add unit tests to cover expected behavior of mtime package functions, remove `n2m()` function in favor of using the `time` package's millisecond functions. Unit tests were added first to ensure that the slight refactor did not break expected behavior at boundaries. 

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by jrmccluskey** [comment]
Run Go PreCommit

**Comment by jrmccluskey** [comment]
Run GoPortable PreCommit

**Comment by jrmccluskey** [comment]
R: @damccorm @riteshghorse

**Review by damccorm — APPROVED** [review]
LGTM! Left a couple optional suggestions

File: sdks/go/pkg/beam/core/graph/mtime/time_test.go [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+		{
+			"insignificant addition large",
+			Time(1000),
+			999999 * time.Nanosecond,
+			Time(1000),
+		},
+		{
+			"significant addition small",
+			Time(1000),
+			1e6 * time.Nanosecond,
```
[the remark this answers is not in the document]
Comment (jrmccluskey) [reply sdks/go/pkg/beam/core/graph/mtime/time_test.go]: Good suggestion, changed

File: sdks/go/pkg/beam/core/graph/mtime/time_test.go [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+		},
+		{
+			"significant addition large",
+			Time(1000),
+			10 * time.Second,
+			Time(11000),
+		},
+		{
+			"add past max timestamp",
+			MaxTimestamp,
```
[the remark this answers is not in the document]
Comment (jrmccluskey) [reply sdks/go/pkg/beam/core/graph/mtime/time_test.go:57]: Another good boundary case. Added.

**Comment by riteshghorse** [comment]
LGTM!

File: sdks/go/pkg/beam/core/graph/mtime/time_test.go [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+			"add across max boundary",
+			Time(int64(MaxTimestamp) - 10000),
+			10 * time.Minute,
+			MaxTimestamp,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if got, want := test.baseTime.Add(test.addition), test.expOut; got != want {
+				t.Errorf("got time %v, want %v", got, want)
```
Comment (lostluck) [inline sdks/go/pkg/beam/core/graph/mtime/time_test.go:71]: This is the kind of test that warrants having clarified inputs.

Same below for the the other tests.

```suggestion
				t.Errorf("(%v).Add(%v), got time %v, want %v", t.baseTime, test.addition, got, want)
```

**Review by lostluck — CHANGES_REQUESTED** [review]

**Comment by jrmccluskey** [comment]
Should be cleaned up now!

File: sdks/go/pkg/beam/core/graph/mtime/time_test.go [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+			"subtract across min boundary",
+			Time(int64(MinTimestamp) + 10000),
+			10 * time.Minute,
+			MinTimestamp,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			if got, want := test.baseTime.Subtract(test.subtraction), test.expOut; got != want {
+				t.Errorf("(%v).Add(%v), got time %v, want %v", test.baseTime, test.subtraction, got, want)
```
Comment (lostluck) [inline sdks/go/pkg/beam/core/graph/mtime/time_test.go:124]:
```suggestion
				t.Errorf("(%v).Subtract(%v), got time %v, want %v", test.baseTime, test.subtraction, got, want)
```

**Comment by lostluck** [comment]
I just merged in the typo fix suggestion. Will merge in after tests pass again.

**Review by lostluck — APPROVED** [review]

**Pull request merged**